sympy/core/power.py

```diff
@@ -193,39 +193,41 @@ class Pow(Expr):
     +--------------+---------+-----------------------------------------------+
     | z**1         | z       |                                               |
     +--------------+---------+-----------------------------------------------+
-    | (-oo)**(-1)  | 0       |                                               |
+    | 1**z         | 1       | because 1 is considered exact                 |
     +--------------+---------+-----------------------------------------------+
-    | (-1)**-1     | -1      |                                               |
+    | oo**-1       | 0.0     | guards against using oo to make               |
+    | (-oo)**-1    |         | conclusions best made with limits             |
+    +--------------+---------+-----------------------------------------------+
+    | (-1)**oo     | nan     | Because of oscillations in the limit.         |
+    | (-1)**-oo    |         |                                               |
     +--------------+---------+-----------------------------------------------+
     | S.Zero**-1   | zoo     | This is not strictly true, as 0**-1 may be    |
     |              |         | undefined, but is convenient in some contexts |
     |              |         | where the base is assumed to be positive.     |
     +--------------+---------+-----------------------------------------------+
-    | 1**-1        | 1       |                                               |
-    +--------------+---------+-----------------------------------------------+
-    | oo**-1       | 0       |                                               |
-    +--------------+---------+-----------------------------------------------+
     | 0**oo        | 0       | Because for all complex numbers z near        |
-    |              |         | 0, z**oo -> 0.                                |
+    | 0.0**oo      |         | 0, z**oo -> 0                                 |
+    +--------------+---------+-----------------------------------------------+
+    | 0.0**e with  | 0.0     | a finite exponent does not remove             |
+    | 0 < e < oo   |         | any difference from 0 represented by 0.0      |
```

Comment:
I don't like listing floats separately from exact values here. I agree that expressions with floating-point inputs should produce floating-point outputs, but that's a general principle in SymPy. Saying `0**x = 0` but `0.0**x = 0.0` here is redundant and distracting IMO. I also don't like calling 0.0 "a difference from 0". If that's true (I don't think it should be), then `0.0**e` ought not to evaluate at all, since "it could be negative" (but again, I don't actually think this).

Is it generally true herethat the value of the result doesn't change from replacing a rational with an equivalent float, only the type? If not, I think it should be.

Reply:
I reviewed the table value with sympy.live output and note:
```
The following were formerly 0 but now

0.0**3 is 0.0 (and 0.0 permits interpretation as 0 +/- eps so could be + or -)
zoo**-1 is 0.0
oo**-oo is 0.0
(+/-oo)**(-1+I) is 0.0

Also, `(-oo)**(1 + I)` is now zoo instead of `(-1)**(1+I)*zoo`
```
I will have to double check that those table values actually compute as represented.

Reply:
**code to run on this PR to affirm table entries**
```python
var('z b')
fzero = lambda x: c == 0 and x is not S.Zero
assert z**1 == z
assert 1**z is S.One
assert fzero(oo**-1)  # @jkalevi wants a change
assert fzero((-oo)**-1)
assert all((-1)**i is S.NaN for i in (-oo, oo))
assert S.Zero**-1 is zoo
assert all(i**oo is S.Zero for i in (0, 0.0))
assert fzero(0.0**S(3))
assert 0**-oo is zoo
assert all(i**j is S.NaN for i in (-1., 1.) for j in (-oo, oo))
assert fzero(zoo**-1)
assert z**zoo is S.NaN
assert oo**oo is oo
assert fzero(oo**-oo)
assert all((-oo)**i is S.NaN for i in (-oo, oo))
assert all(i**(1 + I) is zoo for i in (-oo,oo))
assert all(fzero(i**(-1 + I)) for i in (-oo,oo))
```